build-script.py

```diff
@@ -250,6 +236,54 @@ def generate_gyb_files(gyb_exec, verbose, add_source_locations, destination=None
     print("Done Generating gyb Files")
 
 
+def make_dir_if_needed(path):
+    if not os.path.exists(path):
+        os.makedirs(path)
+
+
+def clear_gyb_files_from_previous_run(sources_dir, destination_dir):
+    for previous_gyb_gen_file in os.listdir(destination_dir):
+        if previous_gyb_gen_file.endswith(".swift"):
+            gyb_file = os.path.join(
+                sources_dir, previous_gyb_gen_file + ".gyb"
+            )
+            if not os.path.exists(gyb_file):
+                check_call(
+                    ["rm", previous_gyb_gen_file], 
+                    cwd=destination, 
+                    verbose=verbose
+                )
+
+
+# Generate the new .swift files in a temporary directory and only copy them
+# to Sources/SwiftSyntaxBuilder/gyb_generated if they are different than the
+# files already residing there. This way we don't touch the generated .swift
+# files if they haven't changed and don't trigger a rebuild.
+def generate_gyb_files_helper(
+    gyb_exec, verbose, add_source_locations, 
+    temp_files_dir, sources_dir, destination_dir
+):
```

Comment:
Some minor documentation improvements.

`Sources/SwiftSyntaxBuilder/gyb_generated` should probably be `destination_dir`.
`temporary directory` could be further specified as `temp_files_dir`.

Also, super nitpicky: Could we re-arrange the arguments here a little to make `verbose` the last argument and the boolean flag `add_source_locations` the second to last one? It better matches the other functions.